src/rules.rs

```diff
@@ -337,65 +337,59 @@ impl StructDatum {
         let mut clauses = vec![wf];
 
         // Types that are not marked `extern` satisfy IsLocal(TypeName)
-        if !self.binders.value.flags.external {
-            // `IsLocalTy(Ty)` depends *only* on whether the type is marked extern and nothing else
-            let is_local = self.binders.map_ref(|bound_datum| ProgramClauseImplication {
-                consequence: DomainGoal::IsLocal(bound_datum.self_ty.clone().cast()),
-                conditions: Vec::new(),
-            }).cast();
-
-            clauses.push(is_local);
-        } else if self.binders.value.flags.fundamental {
-            // If a type is `extern`, but is also `#[fundamental]`, it satisfies IsLocal
-            // if and only if its parameters satisfy IsLocal
-
-            // Fundamental types must always have at least one type parameter for this rule to
-            // make any sense. We currently do not have have any fundamental types with more than
-            // one type parameter, nor do we know what the behaviour for that should be. Thus, we
-            // are asserting here that there is only a single type parameter until the day when
-            // someone makes a decision about how that should behave.
-            assert_eq!(self.binders.value.self_ty.len_type_parameters(), 1,
+        match self.binders.value.struct_type {
```

Comment:
None of this inner code changed, I just moved it all into the branches of the match.